### PR TITLE
Integration of the OVAL object model into the `combine_ovals.py` script

### PR DESCRIPTION
**Pull request by Honny1**
#### Description:

This PR integrates the use of the OVAL object model into the `combine_ovals.py` file.  To do this, the `build_ovals.py` file is rebuilt and cleaned up, as many functions have been integrated into the object model. 

#### Rationale:

This PR is part of the integration of the OVAL object model into the build system. 

#### Review Hints:

To check the functionality of the changes, you can build content using `./build_product` and run a test suite using the `tox` command. 

Depends on: #11235

### CODE REVIEW

**Comment by jan-cerny** [comment]
@Honny1 https://github.com/ComplianceAsCode/content/pull/11235 has been merged, you're unblocked, please rebase on the top of the latest master branch

**Review by jan-cerny — COMMENTED** [review]
It looks amazing, I like the way the model clicks into combine_ovals.py.

File: build-scripts/combine_ovals.py [hunk 1]
```diff
@@ -2,105 +2,108 @@
...
+    ExceptionMissingObjectReferenceInTest,
+    ValueError,
+)
 
 
 def parse_args():
     p = argparse.ArgumentParser()
     p.add_argument(
-        "--build-config-yaml", required=True, dest="build_config_yaml",
+        "--build-config-yaml",
```
[the remark this answers is not in the document]
Comment (Honny1) [reply build-scripts/combine_ovals.py:30]: Okay, I've moved these changes to at least a separate commit.

**Comment by jan-cerny** [comment]
The CI fail on Fedora latest is now caused by upgrade of the underlying image to f39. The problem has been fixed by https://github.com/ComplianceAsCode/content/pull/11256. Please rebase on the top of the latest master branch to bring the changes in and unblock the job.

**Review by jan-cerny — APPROVED** [review]
I have built RHEL9 content from this PR's branch and from the current upstream master branch and I have compared the built `oval-unlinked.xml` from both builds and I haven't noticed any major offenses, the differences were different order of attributes and explicit setting of some attributes to the default values. In terms of content validity and these changes in the built OVAL are fine and they shouldn't have impact on the meaning of the checks therefore I won't insist on making the artifacts identical.

The fail of the Ansible lint job is caused by upgrade of the job to Fedora 39 which brings a different rule set in the linter and therefore isn't caused by changes in this PR.

In general, I find this integration successful.

**Pull request merged**